### PR TITLE
Timeperiod contacts

### PR DESCRIPTION
**Pull request by SylvestreG**

### CODE REVIEW

File: src/checks/checker.cc [hunk 1]
```diff
@@ -369,11 +369,12 @@ void checker::run(
   nagios_macros macros;
   memset(&macros, 0, sizeof(macros));
   grab_host_macros_r(&macros, hst);
+  std::string tmp;
   get_raw_command_line_r(
     &macros,
     hst->check_command_ptr,
     hst->get_check_command().c_str(),
-    nullptr,
+    tmp,
```
Comment (bouda1) [inline src/checks/checker.cc]: I don't understand this change.

File: src/checks/checker.cc [hunk 2]
```diff
@@ -605,11 +606,12 @@ void checker::run(
   memset(&macros, 0, sizeof(macros));
   grab_host_macros_r(&macros, svc->get_host_ptr().get());
   grab_service_macros_r(&macros, svc);
+  std::string tmp;
   get_raw_command_line_r(
     &macros,
     svc->check_command_ptr,
     svc->get_check_command().c_str(),
-    nullptr,
+    tmp,
```
Comment (bouda1) [inline src/checks/checker.cc]: I don't understand this change

File: src/commands/command.cc [hunk 3]
```diff
@@ -152,11 +152,9 @@ commands::command& commands::command::operator=(commands::command const& right)
  *  @return The processed command line.
  */
 std::string commands::command::process_cmd(nagios_macros* macros) const {
-  char* command_line = NULL;
-  process_macros_r(macros, _command_line.c_str(), &command_line, 0);
-  std::string processed_cmd(command_line);
-  delete[] command_line;
-  return (processed_cmd);
+  std::string command_line;
+  process_macros_r(macros, _command_line.c_str(), command_line, 0);
```
Comment (bouda1) [inline src/commands/command.cc]: maybe process_macros_r could take a string as second argument

File: src/configuration/applier/macros.cc [hunk 4]
```diff
@@ -164,8 +145,8 @@ void applier::macros::_set_macro(
        std::string const& value) {
   if (type >= MACRO_X_COUNT)
     throw (engine_error() << "Invalid type of global macro: " << type);
-  if (!_mac->x[type] || strcmp(_mac->x[type], value.c_str()))
-    string::setstr(_mac->x[type], value);
+  if (_mac->x[type] != value.c_str())
```
Comment (bouda1) [inline src/configuration/applier/macros.cc]: I supppose this is a comparison between two strings. So why using c_str().

File: build/tests/CMakeLists.txt [hunk 5]
```diff
@@ -68,7 +68,7 @@ if (WITH_TESTING)
     # Headers.
     "${TESTS_DIR}/timeperiod/utils.hh"
   )
-  target_link_libraries("ut" "gtest" "cce_core")
+  target_link_libraries("ut" "gtest" "cce_core" "asan")
```
[the remark this answers is not in the document]
Comment (SylvestreG) [reply build/tests/CMakeLists.txt]: sorry it is a leftover...

File: inc/com/centreon/engine/macros/grab.hh [hunk 6]
```diff
@@ -153,13 +153,15 @@ namespace  macros {
...
+  std::string get_duration_sec(T& t, nagios_macros* mac) {
     (void)mac;
 
     // Get duration.
     time_t now(time(nullptr));
     unsigned long duration(now - t.get_last_state_change());
-    return string::dup(duration);
+    std::stringstream ss;
+    ss << duration;
+    return ss.str();
```
[the remark this answers is not in the document]
Comment (SylvestreG) [reply inc/com/centreon/engine/macros/grab.hh]: std::to_string ?

File: src/configuration/applier/macros.cc (hunk 4 above)
Comment (SylvestreG) [reply src/configuration/applier/macros.cc]: my bad, I missed it.

File: src/commands/command.cc (hunk 3 above)
Comment (SylvestreG) [reply src/commands/command.cc]: done

**Pull request merged**